src/Features/CSharp/Portable/QuickInfo/CSharpDiagnosticAnalyzerQuickInfoProvider.cs

```diff
@@ -0,0 +1,182 @@
+            {
+                var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var checkIdObject = semanticModel.GetConstantValue(suppressMessageCheckIdArgument.Expression, cancellationToken);
+                if (checkIdObject.HasValue && checkIdObject.Value is string checkId)
+                {
+                    var position = checkId.IndexOf(':');
+                    var errorCode = position == -1
+                        ? checkId
+                        : checkId.Substring(0, position);
+                    errorCode = errorCode.Trim();
+                    return GetQuickInfoFromSupportedDiagnosticsOfProjectAnalyzers(document, errorCode, suppressMessageCheckIdArgument.Span);
+                }
+            }
+
+            return null;
+        }
+
+        private QuickInfoItem? GetQuickInfoFromSupportedDiagnosticsOfProjectAnalyzers(Document document,
+            string errorCode, TextSpan location)
+        {
+            var infoCache = _diagnosticAnalyzerService.AnalyzerInfoCache;
+            var hostAnalyzers = document.Project.Solution.State.Analyzers;
+            var groupedDiagnostics = hostAnalyzers.GetDiagnosticDescriptorsPerReference(infoCache, document.Project).Values;
+            var supportedDiagnostics = groupedDiagnostics.SelectMany(d => d);
+            var diagnosticDescriptor = supportedDiagnostics.FirstOrDefault(d => d.Id == errorCode);
+            if (diagnosticDescriptor != null)
+            {
+                return CreateQuickInfo(location, diagnosticDescriptor);
+            }
+
+            return null;
+        }
+
+        private static QuickInfoItem CreateQuickInfo(TextSpan location, DiagnosticDescriptor descriptor,
+            params TextSpan[] relatedSpans)
+        {
+            var description =
+                descriptor.Title.ToStringOrNull() ??
+                descriptor.Description.ToStringOrNull() ??
+                descriptor.MessageFormat.ToStringOrNull() ??
+                descriptor.Id;
+            var idTag = !string.IsNullOrWhiteSpace(descriptor.HelpLinkUri)
+                ? new TaggedText(TextTags.Text, descriptor.Id, TaggedTextStyle.None, descriptor.HelpLinkUri, descriptor.HelpLinkUri)
+                : new TaggedText(TextTags.Text, descriptor.Id);
+            return QuickInfoItem.Create(location, sections: new[]
+                {
+                    QuickInfoSection.Create(QuickInfoSectionKinds.Description, new[]
+                    {
+                        idTag,
+                        new TaggedText(TextTags.Punctuation, ":"),
+                        new TaggedText(TextTags.Space, " "),
+                        new TaggedText(TextTags.Text, description)
+                    }.ToImmutableArray())
+                }.ToImmutableArray(), relatedSpans: relatedSpans.ToImmutableArray());
+        }
+    }
+
+    internal static class HelperExtensions
```

Comment:
Can we move these to separate file? We should either make it a nested private type or move it to Extensions folder in Features project.

Reply:
I left it in the QuickInfo folder, because the extensions methods are quite specific. A private nested type for extension methods is not possible due to https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs1109. Related commit: 68319328